trace.go

```diff
@@ -185,11 +219,35 @@ type rawTrace struct {
 	Message string `json:"message"`
 }
 
+// AddUserMessage adds user-friendly message describing the error nature
+func (e *TraceErr) AddUserMessage(formatArg interface{}, rest ...interface{}) {
+	newMessage := fmt.Sprintf(fmt.Sprintf("%v", formatArg), rest...)
+	if len(e.Message) == 0 {
+		e.Message = newMessage
+	} else {
+		e.Message = strings.Join([]string{e.Message, newMessage}, ", ")
+	}
+}
+
+// UserMessage returns user-friendly error message
+func (e *TraceErr) UserMessage() string {
+	if e.Message != "" {
+		return e.Message
+	}
+	return e.Err.Error()
+}
+
+// DebugReport returns develeoper-friendly error report
+func (e *TraceErr) DebugReport() string {
```

Comment:
I see no reason why this would not be a default implementation (i.e. String) for `TraceErr`. You basically output the contents of the object - i.e. what a string implementation would (should) do. The implication is that in order to get at this information a type is required (to make a call to the type-specific `DebugReport`), while it is probably useful in a more generic context.
BTW, you're already doing this with `Traces.String`